### ISSUE TITLE
Dataset Question

### ISSUE THREAD

**Issue by stoneshi-1999** [HumaticsLAB/GTM-Transformer#4]
Thank u for sharing ur dataset and model.

I have a few question on the VISUELLE dataset: What’s the specific meaning of the '0'-'11' and 'day','week','month','year' column in the dataset ? How are these figures calculated? THX in advance🙇‍
![Quicker_20220313_221017](https://user-images.githubusercontent.com/70521393/158063766-f3681706-6401-48a4-83bf-ec6f11941f51.png)
![Quicker_20220313_221003](https://user-images.githubusercontent.com/70521393/158063767-b6195e2f-9821-4480-a8e2-b2753c01f765.png)

**Comment by stoneshi-1999** [comment]
And how to make these .pt file in the  VISUELLE？THX !
![image](https://user-images.githubusercontent.com/70521393/158064143-dd288401-e62f-4695-93cb-b3a395880ba5.png)

**Comment by geriskenderi** [comment]
Hello there,

- '0'-'11' refers to sales on that particular week number after the release date. So '2' would be the sales on the second week after the release date. The sales values have been normalized, which is what gives rise to their decimal format.
- 'day','week','month','year' are *normalized* values of each component of the release date. Those values were obtained by essentially diving each component with its maximum (day/31, week/52, month/12, year/2019).
- The .pt files represent dictionaries that contain the mapping of all categories from textual to a particular label. Because we cannot store strings as tensors in PyTorch, in order to feed the data to our Text embedding module we need to store the attributes using a numerical label and then fetch the corresponding string based on it. 

These types of things are explained in our paper: https://arxiv.org/abs/2109.09824. You can also check out the GTM.py file under models to see how the generated PyTorch dataset class is used in each separate module. I believe I have answered all your questions, so I will be closing this issue. Thanks a lot for your interest in our work.

**Comment by stoneshi-1999** [comment]
Thank u for ur patience🙇‍，I read ur paper it's really awesome. And I am still not clear about some points: 

**(1) Does  '0'-'11' refers to the permanence of an item in the shops?**

![image](https://user-images.githubusercontent.com/70521393/158194839-6b51cbb1-fb77-4b63-a513-124d0e0a3b2e.png)

**(2) How to make the 'sales time seires' data ? What's the kind of normalization applied on the sales values?  Because I wanna try to apply GTM in my dataset.** 

**(3) What's the meaning of the output? Is each dimension refers to the forecasted trend of sales on that particular  week?**

![image](https://user-images.githubusercontent.com/70521393/158198283-d4da7c3d-d289-4d65-aacd-057f4ce0fd18.png)

![输出的tensor形状是1，12](https://user-images.githubusercontent.com/70521393/158199037-1e9cd4d4-4335-464a-a921-784f4f252b47.png)

this is the tensor of ' forecast.view(-1, self.output_len) ' ,

![image](https://user-images.githubusercontent.com/70521393/158199944-d2962d1e-be13-4bfe-861f-70c4a68d53a9.png)

**Comment by geriskenderi** [comment]
Hi again,

1. There seems to have been a mistake there, each number refers to the sales of that particular week in all shops of the product. Thanks a lot for pointing it out. As I said before, sales are normalized by the global maximum sold amount of the training set. So if you have a sales dataset of your own, all you have to is calculate the max sold amount from your trainset and divide all the values by that number (Maximum scaling).
2. Yes, with GTM we tried a generative approach to forecasting where each latent dimension actually represents a timestep in the final series. This means the final output tensor of shape B x 12 contains the forecasts for each item in the batch. As mentioned in the paper, we actually consider up to six weeks.

I hope everything is clear now 👍
